pkg/middlewares/metrics/metrics.go

```diff
@@ -81,17 +80,14 @@ func (m *metricsMiddleware) ServeHTTP(rw http.ResponseWriter, req *http.Request)
 	labels := []string{"method", getMethod(req), "protocol", getRequestProtocol(req)}
 	labels = append(labels, m.baseLabels...)
 
-	openConns := atomic.AddInt64(&m.openConns, 1)
-	m.openConnsGauge.With(labels...).Set(float64(openConns))
-	defer func(labelValues []string) {
-		openConns := atomic.AddInt64(&m.openConns, -1)
-		m.openConnsGauge.With(labelValues...).Set(float64(openConns))
-	}(labels)
+	m.openConnsGauge.With(labels...).Add(1)
 
 	start := time.Now()
 	recorder := &responseRecorder{rw, http.StatusOK}
 	m.next.ServeHTTP(recorder, req)
 
+	m.openConnsGauge.With(labels...).Add(-1)
```

Comment:
I've got a strange behaviour with request being rate limited by traefik.

It's like they are not passing here so the gauge is not decremented.